ogr/services/pagure.py

```diff
@@ -140,41 +151,72 @@ def pr_create(self, title, body, target_branch, source_branch):
     def fork_create(self):
         return self.pagure.create_fork()
 
-    @property
-    def fork(self):
-        """PagureRepo instance of the fork of this repo."""
-        kwargs = replace_username_with_username(self.pagure_kwargs)
+    def get_fork(self):
+        """
+        PagureRepo instance of the fork of this repo.
+        """
+        kwargs = self.pagure_kwargs.copy()
         kwargs.update(
             repo=self.repo,
             namespace=self.namespace,
-            username=self.username,
             instance_url=self.instance_url,
             token=self.token,
             is_fork=True,
         )
+        if "username" not in kwargs:
+            kwargs["username"] = self.service.user.get_username()
+
         fork_project = PagureProject(**kwargs)
         try:
-            if fork_project.exists and fork_project.pagure.parent:
+            if fork_project.exists() and fork_project.pagure.get_parent():
                 return fork_project
         except:
             return None
         return None
 
-    @property
-    def is_fork(self):
-        return self._is_fork
-
-    @property
     def exists(self):
-        return self.pagure.project_exists
+        return self.pagure.project_exists()
 
-    @property
     def is_forked(self):
-        return self.fork is not None
+        pass
 
     @property
-    def git_urls(self):
-        return self.pagure.git_urls
+    def is_fork(self):
+        return "fork" in self.namespace
```

Comment:
cheezy

Reply:
I know. The fork/namespace/user logic will probably change when we start implementing other services.

Reply:
no worries, keep it as it is; each implementation can override it easily